### PR TITLE
Portuguese Translations for filter view drop downs

### PR DESCRIPTION
**Pull request by nanoMichal**
Support for Filter translations
Booleans are translatable.. 
Screen Shot 2021-01-04 at 5 20 39 PM
Date Filter is translatable
Screen Shot 2021-01-04 at 5 20 20 PM

Integer Search View also has support

Screen Shot 2021-01-04 at 5 40 55 PM

### CODE REVIEW

File: src/foam/u2/TextField.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ foam.CLASS({
   ],
 
   css: `
-    input[type="search"] {
+    input[type="wtf"] {
```
Comment (nanoMichal) [inline src/foam/u2/TextField.js]:
```suggestion
    input[type="search"] {
```

File: src/foam/u2/filter/properties/DateFilterView.js [hunk 2]
```diff
@@ -15,6 +15,13 @@ foam.CLASS({
     'foam.mlang.Expressions'
   ],
 
+  messages: [
+    { name: 'LABEL_ALL',    message: 'Search' },
```
Comment (nanoMichal) [inline src/foam/u2/filter/properties/DateFilterView.js]:
```suggestion
    { name: 'LABEL_ALL',    message: 'All Time' },
```

File: src/foam/u2/filter/properties/DateFilterView.js [hunk 3]
```diff
@@ -15,6 +15,13 @@ foam.CLASS({
...
+    { name: 'LABEL_AFTER',    message: 'Search' },
```
Comment (nanoMichal) [inline src/foam/u2/filter/properties/DateFilterView.js]:
```suggestion
    { name: 'LABEL_AFTER',    message: 'After' },
```

File: src/foam/u2/filter/properties/DateFilterView.js [hunk 4]
```diff
@@ -15,6 +15,13 @@ foam.CLASS({
...
+    { name: 'LABEL_BEFORE',    message: 'Search' },
```
Comment (nanoMichal) [inline src/foam/u2/filter/properties/DateFilterView.js]:
```suggestion
    { name: 'LABEL_BEFORE',    message: 'Before' },
```

File: src/foam/u2/filter/properties/DateFilterView.js [hunk 5]
```diff
@@ -15,6 +15,13 @@ foam.CLASS({
     'foam.mlang.Expressions'
   ],
 
+  messages: [
+    { name: 'LABEL_ALL',    message: 'Search' },
+    { name: 'LABEL_AFTER',    message: 'After' },
+    { name: 'LABEL_BEFORE',    message: 'Before' },
+    { name: 'LABEL_BETWEEN',    message: 'Search' }
```
Comment (nanoMichal) [inline src/foam/u2/filter/properties/DateFilterView.js]:
```suggestion
    { name: 'LABEL_BETWEEN',    message: 'Between' }
```

File: src/foam/core/ValidationException.js [hunk 6]
```diff
@@ -31,7 +31,7 @@ foam.CLASS({
       name: 'javaExtras',
       buildJavaClass: function(cls) {
         cls.extras.push(foam.java.Code.create({
-          data: `
+        data: `
```
Comment (nanoMichal) [inline src/foam/core/ValidationException.js]:
```suggestion
          data: `
```

File: src/foam/u2/search/TextSearchView.js [hunk 7]
```diff
@@ -82,7 +86,7 @@ foam.CLASS({
     {
       name: 'label',
       expression: function(property) {
-        return property && property.label ? property.label : 'Search';
+        return property && property.label ? property.label : this.LABEL_SEARCH ;
```
Comment (nanoMichal) [inline src/foam/u2/search/TextSearchView.js]:
```suggestion
        return property && property.label ? property.label : this.LABEL_SEARCH;
```

**Review by hchoura — APPROVED** [review]

**Comment by nanoMichal** [comment]
For some reason these small translation changes prevent localhost from loading. can't figure it out.

File: src/foam/i18n/pt_pt/locales.jrl [hunk 8]
```diff
@@ -311,3 +311,39 @@ p({class:"foam.i18n.Locale",locale:"pt",source:"foam.nanos.crunch.SendNotificati
...
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_CLEAR",target:"Limpe o filtro"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LINK_ADVANCED",target:"Filtros avançados"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LINK_SIMPLE",target:"Filtros simples"})
+
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_FILTER",target:"Filtros"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_RESULTS",target:"Resultados do filtro"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.MESSAGE_ADVANCEDMODE",target:"Usando filtros avançados"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.MESSAGE_VIEWADVANCED",target:"Usando filtros simples"})
+
+p({class:"foam.i18n.Locale",locale:"en",source:"net.nanopay.contacts.Contact.ORGANIZATION.label",target:"Nome da empresa"})
```
Comment (nanoMichal) [inline src/foam/i18n/pt_pt/locales.jrl]:
```suggestion
p({class:"foam.i18n.Locale",locale:"pt",source:"net.nanopay.contacts.Contact.ORGANIZATION.label",target:"Nome da empresa"})
```

File: src/foam/i18n/pt_pt/locales.jrl [hunk 9]
```diff
@@ -311,3 +311,39 @@ p({class:"foam.i18n.Locale",locale:"pt",source:"foam.nanos.crunch.SendNotificati
...
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_CLEAR",target:"Limpe o filtro"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LINK_ADVANCED",target:"Filtros avançados"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LINK_SIMPLE",target:"Filtros simples"})
+
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_FILTER",target:"Filtros"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.LABEL_RESULTS",target:"Resultados do filtro"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.MESSAGE_ADVANCEDMODE",target:"Usando filtros avançados"})
+p({class:"foam.i18n.Locale",locale:"pt",source:"foam.u2.filter.FilterView.MESSAGE_VIEWADVANCED",target:"Usando filtros simples"})
+
+p({class:"foam.i18n.Locale",locale:"pt",source:"net.nanopay.contacts.Contact.ORGANIZATION.label",target:"Nome da empresa"})
```
Comment (nanoMichal) [inline src/foam/i18n/pt_pt/locales.jrl]:
```suggestion
```

**Review by jlhughes — APPROVED** [review]

**Pull request merged**